README.md

```diff
@@ -73,13 +73,77 @@ client.search('cat', options).then((resultList) => {
 
 Note that `options` supports all options listed here: https://swiftype.com/documentation/app-search/guides/searching.
 
+### Click Through Tracking
+
+```javascript
+client.click(
+  'Cat',
+  '1234567',
+  '8b55561954484f13d872728f849ffd22',
+  ['Animal']
+).catch((error) => {
+  console.log(`error: ${error}`)
+})
+```
+
+Click throughs can be tracked by binding `client.click` calls to click events on individual search result links.
+
+The following example shows how this can be implemented declaratively by annotating links with class and data attributes.
+
+```javascript
+document.addEventListener('click', function(e) {
+  const el = e.target;
+  if !(el.classList.contains('track-click')) return;
+
+  client.click(
+    el.getAttribute('data-query'),
+    el.getAttribute('data-document-id'),
+    el.getAttribute('data-request-id'),
+    [el.getAttribute('data-tag')]
+  )
+})
+```
+
+```html
+<a href="/items/1234567"
+    class="track-click"
+    data-request-id="8b55561954484f13d872728f849ffd22"
+    data-document-id="1234567"
+    data-query="Cat"
+    data-tag="Animal">
+  Item 1
+</a>
+```
+
 ## Build
 
 ```bash
 yarn install
 yarn build
 ```
 
+## Running Tests
+
+  $ npm test
+
+## Adding an updating Tests
+
+  The specs in this project use [node-replay](https://github.com/assaf/node-replay) to capture responses. The engine which was used to capture these
+  responses was created using the engine created in [this](https://github.com/swiftype/app-search-demo-react#push-data-to-the-node-modules-engine) tutorial. They were then swapped out with fake credentials as to not expose them publicly.
```

Comment:
Not sure I understand this line, the credentials in the spec are real.

```
$ npm test

 PASS  tests/result_item.spec.js
 PASS  tests/client.spec.js

Test Suites: 2 passed, 2 total
Tests:       13 passed, 13 total
Snapshots:   3 passed, 3 total
Time:        2.479s, estimated 7s
Ran all test suites.
```

Reply:
The credentials in this spec are bogus. Meaning none of this is real:
```
const accountHostKey = "host-01kd84";
const apiKey = "search-aw9sk38fjs7akd9ajbnfav69";
const engineName = "test-engine";
```

I just keyed them in to look like real credentials. I'll reword that section to attempt to make it actually make sense. 

That whole process is stupid and clunky. I wonder if we could just create a permanent test instance and just keep real read-only credentials for tests. Or do you think that would be insecure?

I was thinking it might be nice to have something like that anyway, for tutorials and what not. So instead of making a user spin up a new engine, just point them to an existing engine with a read-only key. 

What do you think?

Reply:
Ahh, right, `replay`. I think this setup is great. It allows you to code against a snapshot of the `v1` api that won't be changing in unexpected ways, and it prevents having to hit a real production instance.
Some of the other clients like https://github.com/swiftype/swiftype-app-search-java implicitly require you to enter real credentials to hit a real production instance, see https://github.com/swiftype/swiftype-app-search-java/blob/master/src/test/java/com/swiftype/appsearch/ClientTest.java#L28 which can be nice, but I think it's extreme for an API that shouldn't be changing.

As far as pointing people to an existing engine - we have done that before with "curated data" inside the dashboard, so that you could start playing around with search immediately. However, we have decided that we don't need that for App Search for now, since both indexing and searching are super easy, and we want people to start using the API/clients right away.

Reply:
OK. Yeah, just trying to think about how to make this spec work in a way that if you want to capture new response for replay, it doesn't take a lot of effort. The only thing I can think of is creating a new App Search instance and putting the real (read-only) credentials into this spec. So capturing new responses is easy. Right now, as you can see in the README, it takes a bit of effort.